In [698]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [699]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as mtp
import sklearn.tree as tree
import sklearn.model_selection as ms
import sklearn.ensemble as ens
import sklearn.metrics as met
rf = ens.RandomForestClassifier(n_estimators=100)

In [700]:
#import data
data = pd.read_excel('/content/drive/MyDrive/Colab Notebooks/Data Bank.xlsx')
data

Unnamed: 0          Unnamed: 1  \
0          NaN                 NaN   
1          NaN                 NaN   
2          NaN                 NaN   
3          NaN                 NaN   
4          NaN   Data Description:   
5          NaN                 NaN   
6          NaN                  ID   
7          NaN                 Age   
8          NaN          Experience   
9          NaN              Income   
10         NaN             ZIPCode   
11         NaN              Family   
12         NaN               CCAvg   
13         NaN           Education   
14         NaN            Mortgage   
15         NaN       Personal Loan   
16         NaN  Securities Account   
17         NaN          CD Account   
18         NaN              Online   
19         NaN          CreditCard   

                                           Unnamed: 2  
0                                                 NaN  
1                                                 NaN  
2                                                 NaN  
3                                                 NaN  
4                                                 NaN  
5                                                 NaN  
6                                         Customer ID  
7                   Customer's age in completed years  
8                   #years of professional experience  
9                Annual income of the customer ($000)  
10                             Home Address ZIP code.  
11                        Family size of the customer  
12     Avg. spending on credit cards per month ($000)  
13  Education Level. 1: Undergrad; 2: Graduate; 3:...  
14             Value of house mortgage if any. ($000)  
15  Did this customer accept the personal loan off...  
16  Does the customer have a securities account wi...  
17  Does the customer have a certificate of deposi...  
18  Does the customer use internet banking facilit...  
19  Does the customer use a credit card issued by ...

In [701]:
data.dropna(how="all", inplace=True)
data

Unnamed: 0          Unnamed: 1  \
4          NaN   Data Description:   
6          NaN                  ID   
7          NaN                 Age   
8          NaN          Experience   
9          NaN              Income   
10         NaN             ZIPCode   
11         NaN              Family   
12         NaN               CCAvg   
13         NaN           Education   
14         NaN            Mortgage   
15         NaN       Personal Loan   
16         NaN  Securities Account   
17         NaN          CD Account   
18         NaN              Online   
19         NaN          CreditCard   

                                           Unnamed: 2  
4                                                 NaN  
6                                         Customer ID  
7                   Customer's age in completed years  
8                   #years of professional experience  
9                Annual income of the customer ($000)  
10                             Home Address ZIP code.  
11                        Family size of the customer  
12     Avg. spending on credit cards per month ($000)  
13  Education Level. 1: Undergrad; 2: Graduate; 3:...  
14             Value of house mortgage if any. ($000)  
15  Did this customer accept the personal loan off...  
16  Does the customer have a securities account wi...  
17  Does the customer have a certificate of deposi...  
18  Does the customer use internet banking facilit...  
19  Does the customer use a credit card issued by ...

In [702]:
data.drop("Unnamed: 0",axis = "columns")

Unnamed: 1                                         Unnamed: 2
4    Data Description:                                                NaN
6                   ID                                        Customer ID
7                  Age                  Customer's age in completed years
8           Experience                  #years of professional experience
9               Income               Annual income of the customer ($000)
10             ZIPCode                             Home Address ZIP code.
11              Family                        Family size of the customer
12               CCAvg     Avg. spending on credit cards per month ($000)
13           Education  Education Level. 1: Undergrad; 2: Graduate; 3:...
14            Mortgage             Value of house mortgage if any. ($000)
15       Personal Loan  Did this customer accept the personal loan off...
16  Securities Account  Does the customer have a securities account wi...
17          CD Account  Does the customer have a certificate of deposi...
18              Online  Does the customer use internet banking facilit...
19          CreditCard  Does the customer use a credit card issued by ...

In [703]:
#Prepocessing Data dengan Label Encoder
from sklearn.preprocessing import LabelEncoder
enc = LabelEncoder()

In [704]:
data['Unnamed: 1']=enc.fit_transform(data['Unnamed: 1'].values)
data['Unnamed: 2']=enc.fit_transform(data['Unnamed: 2'].values)
data

Unnamed: 0  Unnamed: 1  Unnamed: 2
4          NaN           4          14
6          NaN           8           3
7          NaN           0           4
8          NaN           6           0
9          NaN           9           1
10         NaN          14          12
11         NaN           7          11
12         NaN           1           2
13         NaN           5          10
14         NaN          10          13
15         NaN          12           5
16         NaN          13           7
17         NaN           2           6
18         NaN          11           9
19         NaN           3           8

In [705]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 15 entries, 4 to 19
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  0 non-null      float64
 1   Unnamed: 1  15 non-null     int64  
 2   Unnamed: 2  15 non-null     int64  
dtypes: float64(1), int64(2)
memory usage: 480.0 bytes


In [706]:
#Import Library
from sklearn.datasets import load_iris
X, y = load_iris(return_X_y=True)
print(f'Dimensi Feature: {X.shape}')
print(f'Class: {set(y)}')

Dimensi Feature: (150, 4)
Class: {0, 1, 2}


In [707]:
#Data Training dan Test
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.6)


In [708]:
#Classification dengan RandomForestClassifier
from sklearn.ensemble import RandomForestClassifier
model = RandomForestClassifier(n_estimators=100, random_state=0)
model.fit(X_train, y_train)

RandomForestClassifier(random_state=0)

In [709]:
X_train

array([[5.1, 3.7, 1.5, 0.4],
       [5.5, 3.5, 1.3, 0.2],
       [5.1, 3.4, 1.5, 0.2],
       [6.7, 3. , 5. , 1.7],
       [4.5, 2.3, 1.3, 0.3],
       [5.8, 2.7, 3.9, 1.2],
       [5.1, 3.8, 1.5, 0.3],
       [5.4, 3.4, 1.7, 0.2],
       [5. , 3.5, 1.3, 0.3],
       [5.7, 3. , 4.2, 1.2],
       [4.6, 3.4, 1.4, 0.3],
       [6.2, 2.9, 4.3, 1.3],
       [6.3, 2.5, 4.9, 1.5],
       [5. , 3.2, 1.2, 0.2],
       [7.7, 2.8, 6.7, 2. ],
       [6.4, 3.1, 5.5, 1.8],
       [7. , 3.2, 4.7, 1.4],
       [5.7, 2.5, 5. , 2. ],
       [6.9, 3.1, 4.9, 1.5],
       [4.7, 3.2, 1.6, 0.2],
       [6. , 2.2, 4. , 1. ],
       [6.8, 2.8, 4.8, 1.4],
       [6.3, 2.3, 4.4, 1.3],
       [6.1, 3. , 4.6, 1.4],
       [5.7, 3.8, 1.7, 0.3],
       [5.8, 2.7, 4.1, 1. ],
       [5.7, 2.8, 4.5, 1.3],
       [7.7, 2.6, 6.9, 2.3],
       [5.8, 2.8, 5.1, 2.4],
       [6. , 2.7, 5.1, 1.6],
       [4.9, 3. , 1.4, 0.2],
       [5.5, 2.5, 4. , 1.3],
       [4.8, 3.1, 1.6, 0.2],
       [5. , 3.5, 1.6, 0.6],
       [4.8, 3

In [710]:
y_train

array([0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 1, 0, 2, 2, 1, 2, 1, 0, 1, 1,
       1, 1, 0, 1, 1, 2, 2, 1, 0, 1, 0, 0, 0, 2, 2, 1, 1, 1, 2, 1, 1, 1,
       0, 0, 0, 0, 2, 2, 0, 0, 2, 2, 1, 2, 2, 1, 2, 1])

In [711]:
X_test

array([[6.4, 3.2, 5.3, 2.3],
       [5.1, 3.5, 1.4, 0.2],
       [5.9, 3. , 4.2, 1.5],
       [6.1, 2.8, 4.7, 1.2],
       [5.8, 2.6, 4. , 1.2],
       [6.2, 2.8, 4.8, 1.8],
       [4.8, 3.4, 1.9, 0.2],
       [5.2, 3.4, 1.4, 0.2],
       [6.7, 2.5, 5.8, 1.8],
       [6.3, 2.7, 4.9, 1.8],
       [6. , 2.9, 4.5, 1.5],
       [7.1, 3. , 5.9, 2.1],
       [6.6, 3. , 4.4, 1.4],
       [7.2, 3. , 5.8, 1.6],
       [5.6, 3. , 4.5, 1.5],
       [6.1, 2.9, 4.7, 1.4],
       [5.5, 4.2, 1.4, 0.2],
       [5.3, 3.7, 1.5, 0.2],
       [5.9, 3. , 5.1, 1.8],
       [6.9, 3.1, 5.1, 2.3],
       [5. , 3. , 1.6, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [6.3, 3.4, 5.6, 2.4],
       [6.4, 3.2, 4.5, 1.5],
       [7.7, 3.8, 6.7, 2.2],
       [5.5, 2.6, 4.4, 1.2],
       [4.9, 2.5, 4.5, 1.7],
       [6.1, 3. , 4.9, 1.8],
       [6.9, 3.1, 5.4, 2.1],
       [5.7, 2.6, 3.5, 1. ],
       [5.1, 3.3, 1.7, 0.5],
       [4.9, 3.1, 1.5, 0.1],
       [4.9, 3.6, 1.4, 0.1],
       [6.7, 3. , 5.2, 2.3],
       [5.7, 4

In [712]:
y_test

array([2, 0, 1, 1, 1, 2, 0, 0, 2, 2, 1, 2, 1, 2, 1, 1, 0, 0, 2, 2, 0, 0,
       2, 1, 2, 1, 2, 2, 2, 1, 0, 0, 0, 2, 0, 0, 2, 2, 2, 1, 1, 0, 2, 0,
       1, 1, 2, 0, 0, 0, 0, 0, 2, 1, 0, 1, 0, 1, 0, 0, 0, 2, 1, 1, 2, 2,
       1, 0, 1, 2, 2, 0, 1, 2, 1, 1, 2, 0, 0, 2, 0, 2, 2, 1, 2, 2, 2, 1,
       2, 2])

In [713]:
y_pred

array([2, 1, 1, 2, 1, 1, 1, 1, 1, 0, 0, 2, 2, 0, 2, 1, 2, 2, 2, 1, 0, 0,
       0, 1, 0, 1, 2, 0, 2, 2, 2, 0, 2, 1, 0, 1, 2, 1, 0, 2, 1, 0, 2, 0,
       1, 1, 0, 2, 0, 1, 0, 1, 1, 1, 2, 1, 1, 2, 2, 0, 0, 2, 1, 2, 0, 2,
       0, 1, 0, 1, 1, 1, 2, 0, 0, 2, 0, 0, 1, 1, 2, 1, 2, 0, 2, 1])

In [719]:
#Evaluasi Model
rf.fit(X_train, y_train)
y_prediksi = rf.predict(X_test)
print('akurasi =', met.accuracy_score(y_test, y_prediksi))
print(met.classification_report(y_test, y_prediksi))
print(rf.feature_importances_)

akurasi = 0.9555555555555556
              precision    recall  f1-score   support

           0       1.00      1.00      1.00        29
           1       0.89      0.96      0.93        26
           2       0.97      0.91      0.94        35

    accuracy                           0.96        90
   macro avg       0.95      0.96      0.96        90
weighted avg       0.96      0.96      0.96        90

[0.08329721 0.04145328 0.41202854 0.46322096]


Kesimpulan Dari hasil pengolahan data pada Data Bank.xlsx dengan metode Random Forest, maka diperoleh tingkat akurasi data sebesar 96% dengan precision pada indexs ke 0 sebesar 100%,  precision pada indexs ke 1 sebesar 89% dan precision pada indexs ke 2 sebesar 97%.
Dengan nilai akurasi yang tinggi ini dapat disimpulkan bahwa pemodelan indikator data customer bank tersebut dapat diterapkan dalam meningkatkan pelayanan pada customer maupun untuk memilah calon customer yang memiliki kriteria baik/positif sehingga dapat membantu meningkatkan kinerja pada sebuah bank.